distributed/scheduler.py

```diff
@@ -256,6 +260,125 @@ def versions(self):
         return self._versions
 
 
+class MemoryInfo:
+    """Memory readings on a worker.
+
+    managed
+        Sum of the output of sizeof() for all dask keys held by the worker, both in
+        memory and spilled to disk
+    managed_in_memory
+        Sum of the output of sizeof() for the dask keys held in RAM
+    managed_spilled
+        Sum of the output of sizeof() for the dask keys spilled to the hard drive.
+        Note that this is the size in memory; serialized size may be different.
+    process
+        Total RSS memory measured by the OS on the worker process.
+        This is always exactly equal to managed_in_memory + unmanaged.
+    unmanaged
+        process - managed_in_memory. This is the sum of
+
+        - Python interpreter and modules
+        - global variables
+        - memory temporarily allocated by the dask tasks that are currently running
+        - memory fragmentation
+        - memory leaks
+        - memory not yet garbage collected
+        - memory not yet free()'d by the Python memory manager to the OS
+
+    unmanaged_old
+        Minimum of the "other" memory measures over the last
+        ``distributed.memory.new_to_old_lag`` seconds
```

Comment:
What's the motivation behind tracking the minimum? Wouldn't a max/mean/median be more informative for scheduling decisions?

Reply:
The idea is that, short of memory leaks, most of the difference between the rolling minimum and the current value ("unmanaged recent") should disappear very soon, as long as your new_to_old_lag is configured to be longer than the runtime of a task and long enough to let gc run

Reply:
makes sense. If this assumption is important, we might want to think about something like an automated warning if average runtime of a task is much larger than the configured value. If this is a recurring issue, we could, of course also define some heuristic, similar to the hearbeat interval.
My experience with task runtimes is simply that they can be very large, especially for custom graphs (think 10s of minutes) and finding a suitable value for everyone is probably quite difficult.

Reply:
We could configure it to "since the oldest task started running +X seconds", but it would not work for seceded tasks and for users that have genuinely long-running tasks that take very little RAM mixed with short bursts of memory-hungry ones.